From 18545032c0b4fa8972835f898d776707724cf29a Mon Sep 17 00:00:00 2001
From: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
Date: Mon, 14 Feb 2022 21:21:27 -0600
Subject: [PATCH 1/1] esmi_oob: Remove APIs calling into I3C driver read/write
 APIs
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit

Removes i3c calls from the library as the library is assuming the use of
i3c-dev tools which has read, write driver API calls. However, BMC
driver APIs has only i3ctransfer.

Signed-off-by: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
---
 CMakeLists.txt          |  6 +++---
 src/esmi_oob/esmi_i2c.c | 14 +++++++-------
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93a3db5..c45ab2f 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,7 +111,7 @@ endif ()
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_common.c")
 set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_i2c.c")
-set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_i3c.c")
+#set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_i3c.c")
 set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_cpuid_msr.c")
 set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_mailbox.c")
 set(ESMI_OOB_SRC_LIST ${ESMI_OOB_SRC_LIST} "${SRC_DIR}/esmi_rmi.c")
@@ -162,8 +162,8 @@ install(FILES ${SOURCE_DIR}/include/esmi_oob/esmi_common.h
                                         DESTINATION include)
 install(FILES ${SOURCE_DIR}/include/esmi_oob/esmi_i2c.h
                                         DESTINATION include)
-install(FILES ${SOURCE_DIR}/include/esmi_oob/esmi_i3c.h
-                                        DESTINATION include)
+#install(FILES ${SOURCE_DIR}/include/esmi_oob/esmi_i3c.h
+#                                        DESTINATION include)
 install(FILES ${SOURCE_DIR}/include/esmi_oob/esmi_cpuid_msr.h
                                         DESTINATION include)
 install(FILES ${SOURCE_DIR}/include/esmi_oob/esmi_mailbox.h
diff --git a/src/esmi_oob/esmi_i2c.c b/src/esmi_oob/esmi_i2c.c
index 8e1dfde..ee9c06d 100644
--- a/src/esmi_oob/esmi_i2c.c
+++ b/src/esmi_oob/esmi_i2c.c
@@ -51,7 +51,7 @@

 #include <esmi_oob/esmi_common.h>
 #include <esmi_oob/esmi_i2c.h>
-#include <esmi_oob/esmi_i3c.h>
+/*#include <esmi_oob/esmi_i3c.h>*/

 #define SBRMI_CTRL	0x1
 #define SBRMI_STATUS	0x2
@@ -93,10 +93,10 @@ oob_status_t esmi_oob_read_byte(struct i2c_info info,
         if (NULL == buffer)
                 return OOB_ARG_PTR_NULL;

-	if (info.i2c_addr > 0xff) {
+	/*if (info.i2c_addr > 0xff) {
 		read_i3c_byte(info, cmd, buffer);
 	}
-	else {
+	else */{
 		snprintf(i2c_path, FILEPATHSIZ, "/dev/i2c-%d", info.i2c_bus);
 		fd = open(i2c_path, O_RDWR);
 		if (fd < 0)
@@ -132,12 +132,12 @@ oob_status_t esmi_oob_write_byte(struct i2c_info info,
 	char i2c_path[FILEPATHSIZ];
 	uint16_t flag = I2C_SLAVE;

-	if (info.i2c_addr > 0xff) {
+	/*if (info.i2c_addr > 0xff) {
 		ret = write_i3c_byte(info, cmd, value);
 		if (ret)
 			return ret;
 	}
-	else {
+	else */{
 		snprintf(i2c_path, FILEPATHSIZ, "/dev/i2c-%d", info.i2c_bus);
 		fd = open(i2c_path, O_RDWR);
 	        if (fd < 0)
@@ -342,9 +342,9 @@ oob_status_t esmi_oob_mailbox_xfer(struct i2c_info info,
 {
 	oob_status_t ret;

-	if (info.i2c_addr > 0xff)
+	/*if (info.i2c_addr > 0xff)
 		ret = esmi_oob_i3c_mailbox_xfer(info, msg);
-	else
+	else*/
 		ret = esmi_oob_i2c_mailbox_xfer(info, msg);
 	return ret;
 }
--
2.17.1

